File: app/src/main/java/com/cliniva/enventory/ui/main/MainActivity.java
package com.cliniva.enventory.ui.main;

import android.support.design.widget.BottomNavigationView;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.os.Bundle;
import com.cliniva.enventory.R;
import com.cliniva.enventory.ui.base.BaseActivity;
import com.cliniva.enventory.ui.customer.CustomerFragment;
import com.cliniva.enventory.ui.home.HomeFragment;
import com.cliniva.enventory.ui.inventory.InventoryFragment;
import com.cliniva.enventory.ui.more.MoreFragment;
import com.cliniva.enventory.ui.sales.SalesFragment;
import com.cliniva.enventory.ui.supplier.SupplierFragment;
import com.cliniva.enventory.ui.supplies.SuppliesFragment;

public class MainActivity extends BaseActivity {

    private BottomNavigationView mBottomNavView;

    @Override
    public int getLayoutRes() {
        return R.layout.activity_main;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Bundle bundle = getIntent().getExtras();

        mBottomNavView = findViewById(R.id.bottom_navigation);
        mBottomNavView.setOnNavigationItemSelectedListener(onNavigationItemSelectedListener);

        if (bundle != null){
            onSwitchFragment(bundle.getInt("TRANSACTION_KEY"));
        } else {
            onFragmentTransaction(HomeFragment.getInstance());
        }
    }

    private void onSwitchFragment(int keyCode){
        switch (keyCode){
            case 0:
                onFragmentTransaction(MoreFragment.getInstance());
                mBottomNavView.setSelectedItemId(R.id.menu_more);
                break;
        }
    }

    private BottomNavigationView.OnNavigationItemSelectedListener onNavigationItemSelectedListener
            = menuItem -> {
        switch (menuItem.getItemId()) {
            case R.id.menu_home:
                onFragmentTransaction(HomeFragment.getInstance());
                return true;
            case R.id.menu_sales:
                onFragmentTransaction(SalesFragment.getInstance());
                return true;
            case R.id.menu_inventory:
                onFragmentTransaction(InventoryFragment.getInstance());
                return true;
            case R.id.menu_supplies:
                onFragmentTransaction(SuppliesFragment.getInstance());
                return true;
            case R.id.menu_more:
                onFragmentTransaction(MoreFragment.getInstance());
                return true;
        }
        return false;
    };

    private void onFragmentTransaction(Fragment fragment){
        FragmentManager manager = getSupportFragmentManager();
        FragmentTransaction transaction = manager.beginTransaction();
        transaction.replace(R.id.fragment_container, fragment).commit();
    }
}

File: app/src/main/java/com/amirsons/inventory/ui/activity/MainActivity.kt
package com.amirsons.inventory.ui.activity

import android.os.Bundle
import android.os.Handler
import android.widget.Toast
import androidx.fragment.app.Fragment
import com.amirsons.inventory.R
import com.amirsons.inventory.ui.base.BaseActivity
import com.amirsons.inventory.ui.fragment.home.HomeFragment
import com.amirsons.inventory.ui.fragment.inventory.InventoryFragment
import com.amirsons.inventory.ui.fragment.more.MoreFragment
import com.amirsons.inventory.ui.fragment.sales.SalesFragment
import com.amirsons.inventory.ui.fragment.supplies.SuppliesFragment
import com.google.android.material.bottomnavigation.BottomNavigationView
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : BaseActivity() {

    private var activeBottomFragment: Fragment? = null

    private var isQuitApplication = false

    override val contentLayout: Int
        get() = R.layout.activity_main

    private val onNavigationItemSelectedListener = BottomNavigationView.OnNavigationItemSelectedListener {

        when (it.itemId) {

            R.id.menu_home -> activeBottomFragment = addFabFragment(HomeFragment.instance, activeBottomFragment, R.id.fragment_container)

            R.id.menu_sales -> activeBottomFragment = addFabFragment(SalesFragment.instance, activeBottomFragment, R.id.fragment_container)

            R.id.menu_inventory -> activeBottomFragment = addFabFragment(InventoryFragment.instance, activeBottomFragment, R.id.fragment_container)

            R.id.menu_supplies -> activeBottomFragment = addFabFragment(SuppliesFragment.instance, activeBottomFragment, R.id.fragment_container)

            R.id.menu_more -> activeBottomFragment = addFabFragment(MoreFragment.instance, activeBottomFragment, R.id.fragment_container)
        }

        true
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        bottom_navigation.setOnNavigationItemSelectedListener(onNavigationItemSelectedListener)
        bottom_navigation.selectedItemId = R.id.menu_home
    }

    override fun onBackPressed() {

        if (isQuitApplication) {
            finish()
        }

        isQuitApplication = true
        Toast.makeText(this, "Press again to exit", Toast.LENGTH_SHORT).show()

        // after 2 sec disable the app twice click again
        Handler().postDelayed({ isQuitApplication = false }, 2000)
    }
}
